Configure

```diff
@@ -167,7 +167,8 @@ my @gcc_devteam_warn = qw(
     -Wsign-compare
     -Wshadow
     -Wformat
-    -Wtype-limits
+    -Wno-type-limits
+    -Wno-tautological-constant-out-of-range-compare
```

Comment:
Namely things like this:
```c
if (SIZE_MAX > UINT64_MAX)
```
This has been causing a few problems recently. We keep committing code that breaks on 32-bit platforms where `SIZE_MAX != UINT64_MAX` and then fixing it later (see @levitte's PR to fix some of the QUIC code on Windows). Testing against `SIZE_MAX` and `UINT64_MAX` is the right/portable way to do this, but some compilers like to complain about a comparison that can "never" be true (e.g. on 64-bit platforms).

I think we should disable these warnings on the grounds that they're effectively actively encouraging one to write worse code.

Reply:
Wouldn't that specific code be better written as a preprocessor conditional?

```` c
#if UINT64_MAX > SIZE_MAX
````

Reply:
Unfortunately this is not possible because we define these constants in `numbers.h` using an expression based on `sizeof()`. We're actually polyfilling these ourselves and not using the values provided by the OS.

Reply:
On compile error I start seeing this note at the end of the compilation errors with this change:

````
cc1: note: unrecognized command-line option '-Wno-tautological-constant-out-of-range-compare' may have been intended to silence earlier diagnostics
````

If there are no compile errors it just seems to ignore the unrecognised flag and continue anyway.